iOS-sample-app/Showcases/DynamicVariables/DynamicVariablesViewController.swift

```diff
@@ -0,0 +1,76 @@
+import UIKit
+import ContentSquare
+
+// This class showcases how to use dynamic variables to gather additional data about the session,
+// such as which version of an A/B test the user saw, or other relevant user data
+class DynamicVariablesViewController: UIViewController {
+
+    @IBOutlet private weak var buyButton: UIButton!
+    private var currentButtonColor: UIColor = .yellow
+    private var userAge: Int = 0
+
+    override func loadView()
+    {
+        super.loadView()
+
+        // Set the color for the button we want to test
+        buyButton.backgroundColor = currentButtonColor
+        // Retrieve the user age, from his personal account for instance
+        userAge = 24
+
+        self.updateDynamicVariables()
+    }
+
+    // MARK: - UI Actions
+
+    @IBAction private func colorSwitchChanged(_ sender: UISwitch)
+    {
+        // Update button color to perform A/B testing
+        buyButton.backgroundColor = currentButtonColor.isEqual(UIColor.yellow) ? UIColor.green : UIColor.yellow
+        currentButtonColor = buyButton.backgroundColor!
+
+        self.updateDynamicVariables()
+    }
+
+    // MARK: - Dynamic vars
+
+    private func updateDynamicVariables()
```

Comment:
I don't think this approach is the best way to showcase Dynamic Variables so that the client understands well its use. It is not your fault though, as I recon it might not be clearly explained enough in our own documentation.
Case in point, I've been digging into their implementation on the DE/App side, in order to understand how it works to propose an implementation technic to one of our clients - they wanted to be able to differentiate screenviews for product list pages, where the same page was reused for multiple types of products (dairies, vegetables, meat, etc...). A simple approach is to use the screenview name, but I was wondering if we could tell them to use Dynamic Variables instead, although I wasn't sure at that time - but I though as you did.

From what I got from my research though, they are not behaving as we would expect. Indeed, when you set a dynamic variable, it is linked at the session level: for example, we can filter all the sessions where "Button Color" = "yellow" was set, but we don't know if it was set partially during the session, or if it had that value for the whole the time of the session.
What this means is, this feature is only to be used to get session level information (such as the User Age you added, or "user was connected", even though that one is tricky as well, as the user might have connected after using the app for a bit unconnected).

This is of course making the dynvar a lot less useful to the clients. I asked about updating it so that it can be used to mark at the pageview level - or even more finely at the event level -, but for the moment it is still on the backlog. I'll let you confirm this with Romain, as I was talking about this 2+ weeks ago with Luis (his boss), but I am pretty sure about this.

Reply:
In the real world a given user is only ever shown one version of a given A/B test. So the AB test version is actually a session level info and Dynamic Variables are a good way to track it.

Reply:
True, that a good use of it. If so, then we might want to add a variable that shows that, as well as some comment regarding the use of dynamic vars for A/B testing - maybe ask to Mannaïg's team for some insights regarding what to put in that comment and the code it goes with ?

Reply:
I updated the sample with session-level informations